### ISSUE TITLE
Bug: ignore directories seems not working from config file

### ISSUE THREAD

**Issue by MathRobin** [eslint/eslint#18312]
### Environment

Node version: 20.12.1
yarn version: 4.1.1
Local ESLint version: 9.0.0
Global ESLint version: -
Operating System: ubuntu 23.10

### What parser are you using?

Default (Espree)

### What did you do?

<details>
<summary>Configuration</summary>

```
import babelParser from '@babel/eslint-parser';

export default [
  {
    ignores: ['.yarn/'],
    languageOptions: {
      parser: babelParser,
      parserOptions: {
        requireConfigFile: false,
        sourceType: 'module',
      },
    },
  },
];

```
</details>

```shell
eslint . -c eslint.config.js
```

### What did you expect to happen?

Nothing

### What actually happened?

```
/my-repo/.yarn/releases/yarn-4.1.1.cjs
  2:1  warning  Unused eslint-disable directive (no problems were reported)

✖ 1 problem (0 errors, 1 warning)
  0 errors and 1 warning potentially fixable with the `--fix` option.
```

### Link to Minimal Reproducible Example

https://github.com/MathRobin/eslint-problem

### Participation

- [ ] I am willing to submit a pull request for this issue.

### Additional comments

if i do
```shell
eslint -c eslint.config.js --ignore-pattern ".yarn/*"
```
it works well.

Looks to repo, all is documented in readme. And in scripts in package.json

**Comment by mdjermanovic** [comment]
If you want ESLint to ignore some files or directories completely, the `ignores` key should be used without any other keys in the configuration object:

```js
import babelParser from '@babel/eslint-parser';

export default [
  {
    ignores: ['.yarn/'],
  },
  {
    languageOptions: {
      parser: babelParser,
      parserOptions: {
        requireConfigFile: false,
        sourceType: 'module',
      },
    },
  },
];
```

https://eslint.org/docs/latest/use/configure/ignore

**Comment by MathRobin** [comment]
Oh great, it's working ! Many thanks ! :heart: 

Not explained into the docs, and mostly : the opposite is shown id doc:
https://eslint.org/docs/latest/use/configure/configuration-files#excluding-files-with-ignores
![2024-04-10_20-11](https://github.com/eslint/eslint/assets/1046585/9340fd3b-599b-46ac-908e-cb2dc7a34a25)

**Comment by mdjermanovic** [comment]
It's different. `ignores` with other keys means that the configuration object doesn't apply to those files/directories:

https://eslint.org/docs/latest/use/configure/configuration-files#excluding-files-with-ignores

`ignores` with no other keys means that those files/directories are completely ignored:

https://eslint.org/docs/latest/use/configure/ignore
